### PR TITLE
New Staking Tokens Guide

### PR DESCRIPTION
**Pull request by TheChronicMonster**
This guide was created based on a tweet thread provided by Austin at xBacked. 
He's credited at the top of the guide and provided permission to turn his thread into a guide for our docs.

Updated the /guide/ seclink and placed this guide in-between the abstracts and ganache guides.

### CODE REVIEW

File: docs/src/guide/staking/index.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+Reach makes DApp development safer, faster, and more secure.
+This guide offers a real example of the advantages of using Reach.
+
+Here's an example provided by Austin Wilshire at [xBacked](https://twitter.com/xbacked). 
+Follow Austin on Twitter [@awoldes](https://twitter.com/awoldes).
+
+Imagine you want to stake a token and at a later date unstake it with all the rewards.
+Let's look at a snippet from the perspective of a Deployer. 
+
+As the deployer, you want to set the reward rate, ASA ID of the token being stakes, the ASA ID of the rewarded token, and the initial supply of the reward token.
```
Comment (jeapostrophe) [inline docs/src/guide/staking/index.md:12]: We can't say "ASA ID". We'd have to say `{!rsh} Token` id

Comment (jeapostrophe) [reply docs/src/guide/staking/index.md:12]: Also, typo on this line, "staked"

File: docs/src/guide/staking/index.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+``` rsh
+.api(User.stakeTokens,
+  (tokenAmt) => {
+      // validation that a user must stake more than 1 token
+      // they will receive the message on the frontend if it fails
+      assume(tokenAmt >= 1, 'Must stake at least 1 token');
+  },
+  (tokenAmt) => {
+      //looks unintuitive, but this means the user will
+      // pay 0 algos, and pay tokenAmt in rewardTokens
```
Comment (jeapostrophe) [inline docs/src/guide/staking/index.md:48]: Can't say "algos", need to say network tokens

File: docs/src/guide/staking/index.md [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+)
+```
+
+## Connect to View
+
+Now for the cool part: Use your contract to hook to your website.
+
+``` rsh
+// connect to the contract
+const application = account.contract(backend, APPLICATION_ID);
```
Comment (jeapostrophe) [inline docs/src/guide/staking/index.md:104]: Can't say application id, need to say contract info

File: docs/src/guide/staking/index.md [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+// defines all User api calls (stakeTokens, unstaketokens)
+const userApi = contract.a.User;
+// this is a view; read the reach docs at docs.reach.sh
```
Comment (jeapostrophe) [inline docs/src/guide/staking/index.md:107]: This should be an actual link (by breaking up the code and/or relying on the automatic code linking)

File: docs/src/guide/staking/index.md [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+  </button>
+</div>
+```
+
+## Conclusion
+
+This demonstrates the essential (but not quite complete) elements of implementing a contract in Reach.
+There's no magic involved, but it definitely feels like there is when you're building a Reach application. 
+As a developer, you don't need to think about forming transaction groups, SDK calls, or opting in. 
+That's right, Reach handles ASA and Application opt-ins. You focus on the business logic of your DApp and plug into the consensus network of your choice.
```
Comment (jeapostrophe) [inline docs/src/guide/staking/index.md:133]: "On a network that requires opt-ins..."

File: docs/src/guide/staking/index.md [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+
+In addition to all of this, Reach also provides formal verification and cross chain compatability. 
+At the time of this writing, Reach currently supports contract deployment on EVM (Ethereum Virtual Machine), AVM (Algorand Virtual Machine), and Conflux consensus networks.
```
Comment (jeapostrophe) [inline docs/src/guide/staking/index.md:136]: Worried about this line getting out of date over time

**Review by jeapostrophe — CHANGES_REQUESTED** [review]

File: docs/src/guide/staking/index.md (hunk 6 above)
Comment (TheChronicMonster) [reply docs/src/guide/staking/index.md]: Removed the final paragraph to keep the guide focused on staking.

File: docs/src/guide/staking/index.md (hunk 1 above)
Comment (hamirmahal) [reply docs/src/guide/staking/index.md]: I recommend [Code Spell Checker](https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker) if you're using VSCode to edit the Markdown, JP.

**Pull request closed without merging**